llvm/lib/CodeGen/DeadMachineInstructionElim.cpp

```diff
@@ -165,10 +180,8 @@ bool DeadMachineInstructionElimImpl::eliminateDeadMI(MachineFunction &MF) {
 
     // Reserved registers are considered always live, so consider them as
     // live-outs for MBB. Inside MBB, dead assignments can still be detected.
-    for (MCPhysReg PhysReg : MRI->getReservedRegs().set_bits()) {
-      if (MRI->canSimplifyPhysReg(PhysReg)) {
-        LivePhysRegs.addReg(PhysReg);
-      }
```

Comment:
What does addLiveOuts do? I can imagine it collects liveOuts from the successor blocks, but since it may not have visited successors yet, these may not be complete. For the unvisited blocks it should add the conservative set, and I hope that includes the simpliiable ones. That would be the way to get the control registers live at function return

Reply:
This uses the liveins from the successor blocks, those are there for each MBB (the virtregrewriter inserts them when dropping the "live range" view). But those do not contain reserved registers.